### PR TITLE
Support adding customized jars after pip install; fix release related

### LINKED ISSUE #1989: support loading custom jars and python-api.zip for python via pip install
When user define new criterions, layers and wrap it in python, it requires loading external jar/python-api.zip when jupyter/BigDL starts up.  But pip install doesn't allow loading customized jars/zip now. External jars can only be loaded using script to launch jupyter. 

We need 

- Support loading custom jars and python-api.zip after pip installation of BigDL
- It's better to have external jars/zip loaded dynamically with a command (which can be executed inside jupyter) or an api.  It won't be convinient if it requires frequent install/uninstall.

### PR DESCRIPTION
**Pull request by hkvision**
## What changes were proposed in this pull request?

Support adding customized jars after pip install.
Fix release issues.
Add docs.

## How was this patch tested?

Jenkins and manual test.

## Related links or issues (optional)
fixed #1989

### CODE REVIEW

File: pyspark/bigdl/util/engine.py [hunk 1]
```diff
@@ -83,13 +87,13 @@ def get_bigdl_classpath():
...
-    if(os.getenv("BIGDL_CLASSPATH")):
-        return os.environ["BIGDL_CLASSPATH"]
+    if os.getenv("BIGDL_CLASSPATH"):
+        return [os.environ["BIGDL_CLASSPATH"]]
     jar_dir = os.path.abspath(__file__ + "/../../")
     jar_paths = glob.glob(os.path.join(jar_dir, "share/lib/*.jar"))
     if jar_paths:
         assert len(jar_paths) == 1, "Expecting one jar: %s" % len(jar_paths)
-        return jar_paths[0]
+        return [jar_paths[0]] + Configuration.get_extra_jars()
```
Comment (zhichao-li) [inline pyspark/bigdl/util/engine.py]: This change should be move to 
```
   if bigdl_classpath and is_spark_below_2_2():
         append_path("SPARK_CLASSPATH", bigdl_classpath)
```
?

File: pyspark/bigdl/util/common.py [hunk 2]
```diff
@@ -114,11 +116,43 @@ def __str__(self):
...
+    def add_extra_python_modules(packages):
+        """
+        Add extra python modules to sys.path
+        :param packages: a string or a list of strings as python package paths
+        """
+        import six
+        if isinstance(packages, six.string_types):
+            packages = [packages]
+        for package in packages:
+            sys.path.append(package)
```
Comment (zhichao-li) [inline pyspark/bigdl/util/common.py]: `sys.path.insert(0, package)` ?

**Comment by hkvision** [comment]
To add extra jar after pip install:
Call before creating SparkConf:
```
from bigdl.util.common import Configuration
Configuration.add_extra_jars('/home/kai/models-0.1-SNAPSHOT.jar')
```
Tested with code:
```
from bigdl.util.common import Configuration
Configuration.add_extra_jars('/home/kai/models-0.1-SNAPSHOT.jar')
import sys
sys.path.insert(0, 'test-models.zip')

from bigdl.util.common import *
from pyspark import SparkContext
sc = SparkContext.getOrCreate(conf=create_spark_conf())
init_engine()

JavaCreator.set_creator_class("com.intel.analytics.zoo.models.pythonapi.PythonModels")
from customized.zoo.models import *
conf = Configure()
```
cc @zhichao-li @jason-dai @shane-huang

**Comment by zhichao-li** [comment]
LGTM.

**Comment by hkvision** [comment]
Manually tested on `jupyter` and terminal started with `python`.

**Comment by hkvision** [comment]
jenkins passed http://172.168.2.101:8080/view/PR-V-P/job/BigDL-PR-V/667/

**Pull request merged**